server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java

```diff
@@ -961,7 +962,14 @@ private RealmsContext resolveRealms(SecurityType securityConfig, SecurityContext
                             Node node = childNodes.item(i);
                             if (Node.ELEMENT_NODE == node.getNodeType()) {
                                 NodeList content = node.getChildNodes();
-                                options.put(node.getLocalName(), content.item(0).getNodeValue());
+
+                                // Avoid NPE at the startup when an empty option is specified in the config.
+                                if ((content != null) && (content.item(0) != null)) {
+                                    options.put(node.getLocalName(), content.item(0).getNodeValue());
+                                }
+                                else {
+                                    options.put(node.getLocalName(), EMPTY_STRING);
```

Comment:
Why need the empty string here, better to leave out and make the reader tolerant of missing option, same as other options?